src/Engine/Audio/VorbisFile.cpp

```diff
@@ -9,40 +9,43 @@
 
 using namespace Audio;
 
-VorbisFile::VorbisFile(const char* filename) {
+Audio::VorbisFile::VorbisFile()
+{
 
-    // Open OGG file
-    int error;
-    stb_vorbis* stbfile = stb_vorbis_open_filename(filename, &error, NULL);
-    if (error == VORBIS_file_open_failure)
-        Log() << "Couldn't load OGG Vorbis file: " << filename << "\n";
-    else {
-        stb_vorbis_info info = stb_vorbis_get_info(stbfile);
-        const int samples = stb_vorbis_stream_length_in_samples(stbfile) * info.channels;
-        sampleRate = info.sample_rate;
-
-        // Get data
-        data = new float[samples * sizeof(float)];
-        stb_vorbis_get_samples_float_interleaved(stbfile, info.channels, data, samples);
-
-        dataSize = samples;
-        
-        stb_vorbis_close(stbfile);
-    }
 }
 
 VorbisFile::~VorbisFile() {
-
+    stb_vorbis_close(stbFile);
 }
 
-float* VorbisFile::GetData() const {
-    return data;
+int VorbisFile::GetData(uint32_t offset, uint32_t samples, float* data) const {
+    stb_vorbis_seek(stbFile, offset);
+    return stb_vorbis_get_samples_float_interleaved(stbFile, channelCount, data, samples);
 }
 
-uint32_t VorbisFile::GetSize() const {
-    return dataSize;
+uint32_t VorbisFile::GetSampleCount() const {
+    return sampleCount;
 }
 
 uint32_t VorbisFile::GetSampleRate() const {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') The function 'GetSampleRate' is never used. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)